listener.go

```diff
@@ -92,6 +92,18 @@ func (l *listener) setupConn(sess quic.Session) (*conn, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// cancel all active hole punches
+	l.transport.holePunchingMx.Lock()
+	holePunches, ok := l.transport.holePunching[remotePeerID]
```

Comment:
I'd prefer to just cancel the hole punch for this `addr`, not for all others.
The swarm will make sure that we only have a single connection to a peer, but that's a logic that should only live there. From the point of a transport implementation, it's totally fine to have multiple connections to the same peer.

Since the swarm will cancel the context passed to `Dial` as soon as it has one connection, this will also stop the hole punching for all other connection attempts.

I think this will allow you to simplify the map by removing the peer ID entry.

Reply:
Ok, can do.

Reply:
hrm, how do I get the `UDPAddr` from an `Addr`? That's all we have in the quic session.

Reply:
I guess I can use a string, assuming the two map to the same string.

Reply:
it's probably a UDPAddr underneath, but I don't really want to cast, so will use string as keys.

Reply:
done.

Reply:
Yes, that should be fine.